Makefile

```diff
@@ -1,74 +1,68 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-.PHONY: build clean dev help sdist test install remote
+.PHONY: clean dev dev-with-widgets help install js sdist test 
+
+REPO:=jupyter/pyspark-notebook:a388c4a66fd4
```

Comment:
https://github.com/jupyter/docker-stacks#stacks-tags-versioning-and-progress

tl;dr: 4.0 is legacy and no longer moving.